File: Ad-Hoc/__1000__/1136/1136.php
<?php

while(true){

    list($n, $b) = explode(" ", readline());
    $n = (int) $n;
    $b = (int) $b;

    if($n == 0 && $b == 0) break;

    $bolas = explode(" ", readline());
    $possibilidades = [];
    $possibilidades[0] = true;

    for($i = 0; $i < $b; $i++){
        for($j = $i + 1; $j < $b; $j++){
            $diff = abs((int)$bolas[$i] - (int)$bolas[$j]);
            $possibilidades[$diff] = true;
        }
    }

    if(sizeof($possibilidades) == $n + 1){
        echo "Y\n";
    } else {
        echo "N\n";
    }
}

File: Ad-Hoc/__1000__/1136/1136.cpp
#include <iostream>
#include <set>
#include <vector>
#include <cmath>

using namespace std;

int main(){

    int N, B;
    cin >> N >> B;
    while(N != 0 && B != 0){

        vector<int> bolas;
        for(int i = 0; i < B; i++){
            int bola;
            cin >> bola;
            bolas.push_back(bola);
        }

        set<int> possibilidades;
        possibilidades.insert(0);

        for(int i = 0; i < B; i++){
            for(int j = i + 1; j < B; j++){
                possibilidades.insert(abs(bolas[i] - bolas[j]));
            }
        }

        cout << (possibilidades.size() == N + 1 ? 'Y' : 'N') << endl;
        cin >> N >> B;
    }

    return 0;
}